shared/src/main/scala/org/atnos/eff/Interpret.scala

```diff
@@ -581,6 +581,57 @@ trait Interpret {
     }
   }
 
+  type of[F[_], G[_]] = {type l[A] = F[G[A]]}
+
+  /**
+   * Intercept the values for one effect and transform them into
+   * other values for the same effect
+   */
+  def interleaveMonadTransformation[R, T[_], F[_], A](effects: Eff[R, A], nat: T ~> (T `of` F)#l)
```

Comment:
This looks like a net win for me. I would however create one additional method specializing `interleaveMonadTransformation` to the case where `F = Id` and use that simpler method where it can be used.